lib/runner/TestRunner.js

```diff
@@ -175,7 +182,13 @@ module.exports = class TestRunner {
     filterRequest(interaction, request) {
         interaction.applyExpressions(request);
 
+        // We check if a filter object is defined
+        // If so, it gives the test writer a chance to make changes to the request
         const testSuite = interaction.test.testSuite;
+        if (testSuite.filterObject() && testSuite.filterObject().onRequest) {
+            testSuite.filterObject().onRequest(interaction.test, request);
```

Comment:
should this calls support async methods?

Reply:
To date, our filters have always returned void - it has never come up that async operations would happen here. I would want to know of a use-case before we include this.